src/HomeNet/Network/ClientList.cs

```diff
@@ -83,6 +78,25 @@ public List<Client> GetNetworkClientList()
     }
 
 
+    /// <summary>
+    /// Gets ID for a new client.
+    /// </summary>
+    /// <returns>New client's network ID.</returns>
+    public ulong GetNewClientId()
+    {
+      log.Trace("()");
+      ulong res = 0;
+
+      lock (lockObject)

```

Comment:
It's more of a coding habit thing. It most likely won't have any performance issue. But in general it won't hurt to have separated locks neither.